tree/treeplayer/inc/TMPWorkerTree.h

```diff
@@ -73,13 +78,18 @@ class TMPWorkerTree : public TMPWorker {
 template<class F>
 class TMPWorkerTreeFunc : public TMPWorkerTree {
 public:
-   TMPWorkerTreeFunc(F procFunc, const std::vector<std::string>& fileNames,
-                                 const std::string& treeName, unsigned nWorkers, ULong64_t maxEntries)
-                  : TMPWorkerTree(fileNames, treeName, nWorkers, maxEntries),
-                    fProcFunc(procFunc), fReducedResult(), fCanReduce(false) {}
-   TMPWorkerTreeFunc(F procFunc, TTree *tree, unsigned nWorkers, ULong64_t maxEntries)
-                  : TMPWorkerTree(tree, nWorkers, maxEntries),
-                    fProcFunc(procFunc), fReducedResult(), fCanReduce(false) {}
+   TMPWorkerTreeFunc(F procFunc, const std::vector<std::string> &fileNames, TEntryList *entries,
+                     const std::string &treeName, unsigned nWorkers, ULong64_t maxEntries, ULong64_t firstEntry)
+      : TMPWorkerTree(fileNames, entries, treeName, nWorkers, maxEntries, firstEntry), fProcFunc(procFunc),
+        fReducedResult(), fCanReduce(false)
+   {
+   }
+   TMPWorkerTreeFunc(F procFunc, TTree *tree, TEntryList *entries, unsigned nWorkers, ULong64_t maxEntries,
```

Comment:
[I am realize this is pre-existing to this PR]  Should we use a reference to TTree here?  I.e. is tree==nullptr a useful case?